### DISCUSSION TITLE
posts/Go/socket/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#359] [Announcements]
https://www.liwenzhou.com/posts/Go/15_socket/ 

 李文周的Blog Go语言 socket 网络编程 TCP UDP net http go

**Comment by muyids** [comment]
tcp连接建立后，如果server退出，client端应如何处理

**Comment by GGGIsGGG** [comment]
> @muyids
> tcp连接建立后，如果server退出，client端应如何处理

搞个最大响应时间吧，如果超时就自动断开，报错啥的呗。

**Comment by VEH47** [comment]
> @muyids
> tcp连接建立后，如果server退出，client端应如何处理

请了解四次挥手

**Comment by swahpy** [comment]
讲得很好，受益匪浅，谢谢！

**Comment by seabra2** [comment]
为什么udp连接与tcp连接建立方法不太同呢？
